### PR TITLE
Graphical enhancements of line model and IEEE14 example

### PR DESCRIPTION
**Pull request by ceraolo**
I've implemented some graphical enhancements to the library.
Mainly:
1. enlarged the text boxes so that more space is mada available to theri content. The previous text boxes caused nearly always the names to be truncated with ellipises
2. visually differentiated positive and negative pins in lines and transformers. Now, in compliance with the de-facto standard of Modelica Standard library, the positive pins are blue-fitted, the negative ones white fitted..

More details are available in the comments of individual commit composing thsi PR.

### CODE REVIEW

**Comment by dietmarw** [comment]
@ceraolo Could you accept the CLA by following the instructions above so we can use your contribution here? Thanks.

**Comment by ceraolo** [comment]
Dear all,
I'm satd that Modelica tools reformat completely the model texts. 
This PR implies very limited changes, but looking at the individual commits, gitHub shows that nearly everything was changed.

This does not allow reviewers to identify the small changes I've proposed.

I expect a future world in which modelica tools do not change arbitrarily text formatting, so that changes are immediately visible, as normally occurs with all the other languages I know.

**Comment by dietmarw** [comment]
Thanks, @ceraolo for the PR. Before going into the technical details I have some general comments/questions/tips:
1. First of all, don't worry about it looking like a lot of changes. This is most likely down to your PR changing all line endings. Looks like your git-client does not obey the setting we have in `.gitattributes`.  I can easily fix this (and will do so) and force update the PR again so it only shows the real changes. You can also tell GitHub to ignore white-space changes in the display (although ideally one should still not commit those):
  image

2. You created the PR based on the master branch in your fork. This basically means that that master branch is now locked to the development and changes within this PR until it gets finally merged at some point. This usually is very inconvenient (especially for you) if you want to also continue to work on other things and maybe want to create other PRs. The proper way  for the future would be: 
    1. you update your local master to be in sync with the latest status of OpenIPSL/master
    2. You create a new branch and name it something that is meaningful for the planned PR (that's more for yourself than for us so that you can remember what that branch was for again at a later point in time).
    3. you commit your changes and push that branch to your fork and create a PR based on that one.
 3. In e96037224bd551eb63b8c8c58bb040989f9739e6 you added  `GroupBus*.mo` files on the top-level outside the OpenIPSL library itself. I guess what you meant to do is to change `OpenIPSL/Examples/IEEE14/Generation_Groups/GroupBus*.mo` instead. If yes then I can fix this for you. But I just wanted to make sure.

**Comment by ceraolo** [comment]
> 
> 1. First of all, don't worry about it looking like a lot of changes.  This is most likely down to your PR changing [...] 

Wow! This is good news for me.

> 2. You created the PR based on the master branch in your fork. This basically means that that master branch is now locked to the development and changes within this PR until it gets finally merged at some point. This usually is very inconvenient (especially for you) if you want to also continue to work on other things and maybe want to create other PRs. 
You're perfectly right. I'm very new to PRs, and therefore I knew I would make huge mistakes, although I hoped they did not jeopardise the meaning of the PR.

In the past I tried to link Dymola with gitHub (i.e. use gitHub for version control within Dymola) , but found the Dymola documentation very confusing on this point. In the end I got things work well with my own repositories, but I did not trust myself using this way for this PR.
I'll tell you a secret.  What I did is very basic: for each commit I dragged-and-dropped files, and GitHub created commits, letting me specifying commit messages (which I tried to make very clear and complete, though short)

>The proper way  for the future would be: [...]

Thanks. I'll sure follow your recommendations

> 3. In [e960372](https://github.com/OpenIPSL/OpenIPSL/commit/e96037224bd551eb63b8c8c58bb040989f9739e6) you added  `GroupBus*.mo` files on the top-level outside the OpenIPSL library itself. I guess what you meant to do is to change `OpenIPSL/Examples/IEEE14/Generation_Groups/GroupBus*.mo` instead. If yes then I can fix this for you. But I just wanted to make sure.

Those GroupBus files in the top-level appeared by mistake, but my way of making commits (as I said dragging-and-dropping) did not allow me to delete them,. At least I could not find a proper way to do so.
From a message from GitHub I suspected they would not be included in the PR, but now I realise they are. They should be deleted.
I already inserted changes inside OpenIPSL/Examples/IEEE14/Generation_Groups, as you can see in the related commit.

Thanks again for your very insightful comments and suggestions.

**Review by dietmarw — APPROVED** [review]
I've cleaned up this PR. Some notes: 
1. Don't change the diagram size away from the default if possible. I've fixed this.
2. I've improved the addition of the new pins by use of inheritance from the original `PwPin`
3. Changing existing models to use the new `PwPin_p` and `PwPin_n` will have to wait until a new non-backward release is planned. I've collected those changes in the draft PR #337. Most likely there are quite a lot of other models that should be updated. But that can be dealt with in time.

**Comment by dietmarw** [comment]
@marcelofcastro If you have the time, it would be nice if you could take a last look before I merge it. Thanks.

**Comment by marcelofcastro** [comment]
Will do, @dietmarw!

**Review by marcelofcastro — COMMENTED** [review]
@dietmarw it looks good to me. I like the usage of positive/negative pins for lines and transformers, especially to indicate the original purpose of the line ends. I think my question/comment would be with respect to adding a piece of documentation in both transformer and transmission line to indicate that the p and n pins, representing sending and receiving ends in the transmission line, and the primary and secondary sides in the transformer. Do you believe that this is valuable?

**Comment by dietmarw** [comment]
@marcelofcastro Personally I have a problem getting my head around the definition of "sending"  and "receiving". Should p be sending or receiving for example? If you connect a line you would naturally connect the p connector to the generator side in which case it would be receiving? So it's fine for me if additional documentation is added as long as it does not add more confusion. Maybe  @ceraolo can do that.

**Comment by ceraolo** [comment]
> I've cleaned up this PR. Some notes:
> 
> 1. Don't change the diagram size away from the default if possible. I've fixed this.
> 2. I've improved the addition of the new pins by use of inheritance from the original `PwPin`
> 3. Changing existing models to use the new `PwPin_p` and `PwPin_n` will have to wait until a new non-backward release is planned. I've collected those changes in the draft PR #337. Most likely there are quite a lot of other models that should be updated. But that can be dealt with in time.

Just to remind here that indeed no change or conversion script is needed, since the proposal was backards compatible. So no need to wait for major releases. See https://github.com/OpenIPSL/OpenIPSL/pull/337#issuecomment-1412349619

**Comment by dietmarw** [comment]
@ceraolo No need for a reminder. As you can see this PR is scheduled for the next minor version.

**Comment by ceraolo** [comment]
> @marcelofcastro Personally I have a problem getting my head around the definition of "sending" and "receiving". Should p be sending or receiving for example? If you connect a line you would naturally connect the p connector to the generator side in which case it would be receiving? So it's fine for me if additional documentation is added as long as it does not add more confusion. Maybe @ceraolo can do that.

Lines and transformers are reversible: power can flow in both directions.
So the differentiation of positive and negative pins is only to ease identification of pins when the user looks at the simulation outputs, not to say which is sending and receiving.

For additional clarity I show an example hereafter.
Consider the following screenshots (I take as example OpenModelica, but this is similar to other tools).
**Before PR:**
image
**After PR:**
image
If I wanted to understand which pin current p.ii refers to, prior to the change of the appearence, I had to go with the mouse to the two line pins to see which is the positive, what is the negative. And this had to be repeated for every pin voltage or current.
After the PR, it is immediate which pin every current and voltage belongs to.

This is very similar to what occurs in Modelica Standard Library for basic components such as resistors, inductors, capacitors, etc.. Every user might have their own idea of what pin is better to be imagined as "positive" or "negative", (or "sending" and "receiving") but no general rule (wisely) is said in the MSL documentation.

To conclude, I would not recommend writing anything in the documentation regarding the difference between positive and negative pins.

**Comment by marcelofcastro** [comment]
@dietmarw and @ceraolo sending and receiving are just terms to facilitate the reference. In a line, P could be set as sending and N could be set as receiving. But it's just a reference name, they have no physical meaning in the sense of sending or receiving power or current because, of course, power flow is usually reversible in many lines and transformers and because we are talking about AC current. So sending and receiving are just the names.

I was just trying to point out that if we have two different connectors in the library (actually three, but PwPin and PwPin_p are the same) it would be good for the user to understand that there is no real physical difference in them. It is just a matter of reference, right? Same thing for positive/negative pins in resistors, for example. I just thought it could be useful for someone to understand how to use the "different" connectors when a new model is developed and why we created these different connectors. I'd believe that people using the library would ask questions like: Why three connectors instead of just one, since all of them have the same set of equations? Why two of them are exactly the same and the other one has a different color? Transformers and lines have two different connectors, is that a particular rule I should follow when connecting these elements in a circuit?

Anyway, I was just trying to provide more information to a possible user, but if you guys think that no documentation on the connectors or components is the way to go, I'd be ok.

**Comment by ceraolo** [comment]
> 



> (actually three, but PwPin and PwPin_p are the same) it would be good for the user to understand that there is no real physical difference in them. It is just a matter of reference, right?
The simultaneous presence of PwPin and PwPin_p is to avoid breaking models already using PwPin. After all, also in MSL.Analog.Interfaces we have the three pins Pin, Positive Pin, NegativePin.

in MSL I read: 
`Connectors PositivePin and NegativePin are nearly identical. The only difference is that the icons are different in order to identify more easily the pins of a component. Usually, connector PositivePin is used for the positive and connector NegativePin for the negative pin of an electrical component.
`

Maybe for our pins we inspire to this and use:

`Connectors Pin_p and Pin_n are nearly identical. The only difference is that the icons are different in order to identify more easily the pins of a component.
`

or, if you prefer:

`Connectors Pin_p and Pin_n are nearly identical. The only difference is that the icons are different in order to identify more easily the pins of a component. In case a component, e.g. a transformer, is used in a system where it has has a preferential direction of power, usually pin_p and pin_n are the connectors where power usually enters or exits the component, respectively.
`

@marcelofcastro what do you think?

**Comment by marcelofcastro** [comment]
I believe the first option is already enough. As I said before, didn't want to make anything complicated or painful haha Just trying to think what would help someone trying to use what you have just added to the library.

Once you add the description I can review it again and approve the PR.

**Comment by ceraolo** [comment]
Frankly, I'm not sure to be able to make changes to an edited PR like this without making disasters.

@dietmarw May I ask you to add the description to PwPin_p and PwPin_n of this PR?

As you can see, @marcelofcastro agreed that the following description is ok:

`Connectors PwPin_p and PwPin_n are nearly identical. The only difference is that the icons are different in order to identify more easily the pins of a component. `

As in MSL I think it is good to put this description in both PwPin_p and PwPin_n.

Thank you!

**Comment by dietmarw** [comment]
Done. @marcelofcastro please have another look.

**Review by marcelofcastro — APPROVED** [review]
It looks good to me.

**Pull request merged**
